core/shared/src/main/scala/fs2/Pull.scala

```diff
@@ -515,16 +515,23 @@ object Pull extends PullLowPriority {
       fun: O => Pull[F, P, Unit]
   ) extends Action[F, P, Unit]
 
-  /** Steps through the stream, providing either `uncons` or `stepLeg`.
+  /** Steps through the given inner stream, until the first `Output` is reached.
+    * It returns the possible `uncons`.
     * Yields to head in form of chunk, then id of the scope that was active after step evaluated and tail of the `stream`.
     *
     * @param stream             Stream to step
-    * @param scopeId            If scope has to be changed before this step is evaluated, id of the scope must be supplied
     */
-  private final case class Step[+F[_], X](stream: Pull[F, X, Unit], scope: Option[Unique])
-      extends Action[Pure, INothing, Option[StepStop[F, X]]]
+  private final case class StepCons[+F[_], +O](stream: Pull[F, O, Unit])
```

Comment:
Can we call this `Uncons`?

Reply:
The idea occurred to me... was not sure if that would be too different from existing one.